docs/examples/how-tos.md

```diff
@@ -0,0 +1,396 @@
+
+To insert some longer HTML code, you can parse it to the {@link module:engine/model/documentfragment~DocumentFragment model fragment} first and then {@link module:engine/model/model~Model#insertContent insert} it into the editor model:
+
+```js
+const content =
+	'<p>A paragraph with <a href="https://ckeditor.com">some link</a>.</p>';
+const viewFragment = editor.data.processor.toView( content );
+const modelFragment = editor.data.toModel( viewFragment );
+
+editor.model.insertContent( modelFragment );
+```
+
+### How to focus the editor?
+
+```js
+// Focus the editor.
+editor.focus();
+
+// Focus the editing area of the editor.
+editor.editing.view.focus();
+```
+
+### How to delete selected blocks?
+
+```js
+const selectedBlocks = Array.from(
+	editor.model.document.selection.getSelectedBlocks()
+);
+const firstBlock = selectedBlocks[ 0 ];
+const lastBlock = selectedBlocks[ selectedBlocks.length - 1 ];
+
+editor.model.change( writer => {
+	const range = writer.createRange(
+		writer.createPositionAt( firstBlock, 0 ),
+		writer.createPositionAt( lastBlock, lastBlock.maxOffset )
+	);
+
+	writer.remove( range );
+} );
+```
+
+### How to delete all specific elements (e.g. block images) in the editor?
+
+```js
+editor.model.change( writer => {
+	const range = writer.createRangeIn( editor.model.document.getRoot() );
+	const itemsToRemove = [];
+
+	for ( const value of range.getWalker() ) {
+		if ( value.item.is( 'element', 'imageBlock' ) ) {
+			// a different `is` usage.
+			itemsToRemove.push( value.item );
+		}
+	}
+
+	for ( const item of itemsToRemove ) {
+		writer.remove( item ); // remove all of the items.
+	}
+} );
+```
+
+### How to place the caret at the beginning or the end?
+
+```js
+// Place it at the beginning.
+editor.model.change( writer => {
+	writer.setSelection(
+		writer.createPositionAt( editor.model.document.getRoot(), [ 0 ] )
+	);
+} );
+
+// Place it at the end.
+editor.model.change( writer => {
+	writer.setSelection(
+		writer.createPositionAt( editor.model.document.getRoot(), 'end' )
+	);
+} );
+```
+
+### How to find all specific elements (e.g. links) in the editor?
+
+```js
+const range = editor.model.createRangeIn( editor.model.document.getRoot() );
+const items = [];
+
+for ( const value of range.getWalker() ) {
+	if ( value.type === 'text' && value.item.hasAttribute( 'linkHref' ) ) {
+		console.log( value.item );
+	}
+}
+```
+
+### How to listen on a double click (e.g. link elements)?
+
+```js
+// Add observer for double click and extend a generic DomEventObserver class by a native DOM dblclick event:
+import { DomEventObserver } from 'ckeditor5/src/engine';
+
+class DoubleClickObserver extends DomEventObserver {
+	constructor( view ) {
+		super( view );
+
+		this.domEventType = 'dblclick';
+	}
+
+	onDomEvent( domEvent ) {
+		this.fire( domEvent.type, domEvent );
+	}
+}
```

Comment:
I'd mention that there are many observers provided with our features and that you should check if there is no conflicting observer that already fires for the given DOM event.

Reply:
Ok, to fix.